packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1393,21 +1452,23 @@ export class TransactionController extends BaseController<
         delete txParams.gasPrice;
       }
 
-      const unsignedEthTx = this.prepareUnsignedEthTx(txParams);
-      const signedTx = await this.sign(unsignedEthTx, from);
-      await this.updateTransactionMetaRSV(transactionMeta, signedTx);
-      transactionMeta.status = TransactionStatus.signed;
-      this.updateTransaction(
-        transactionMeta,
-        'TransactionController#approveTransaction - Transaction signed',
-      );
+      if (!this.beforePublish(transactionMeta)) {
+        transactionMeta.status = TransactionStatus.approved;
```

Comment:
Is this not already set on line 1428?

Could we instead always update above, and then this condition could go after the standard call to `signTransaction` and then we just `return` inside the block?

Reply:
Done. https://github.com/MetaMask/core/pull/1787/commits/82e9e4dd9604069f0d0bb9abcb284cb724844633